### PR TITLE
[Improve][Transform] Improve inner sql query for map field type

### PR DESCRIPTION
**Pull request by liunaijie**
### Purpose of this pull request

Now the sql transform is support inner query, we can use sql like this
```
select col_a.sub_col_b
```
But for now, if the `sub_col_b`  is map type. will got exception.

improve this feature to support query map type as latest field.

Why query map type got exception:
when parse the sql, we need return the query column's field type.
if the query column is map, get the column type is too complex. so only support query from map type only once.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

update e2e test conf

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/inner_query.conf [hunk 1]
```diff
@@ -53,6 +53,7 @@ transform {
         c_row.c_string c_string,
         c_row.c_inner_row.c_inner_string c_inner_string,
         c_row.c_inner_row.c_inner_timestamp c_inner_timestamp,
+        c_row.c_inner_row.c_map inner_map,
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/inner_query.conf:56]: Please update Assert config.

**Review by Hisoka-X — COMMENTED** [review]
Please update doc.

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/inner_query.conf (hunk 1 above)
Comment (liunaijie) [reply seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/inner_query.conf:56]: Done

**Comment by liunaijie** [comment]
> Please update doc.

截屏2024-09-25 21 27 08

current document not right, we can't query map column now, in this fix we can support that.

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @liunaijie

**Review by hailin0 — APPROVED** [review]

**Pull request merged**